File: movie/lib/model/subcategories_model.dart
class subcategories_model{
  final int id;
  final String title;
  final String image;
  final int categories_id;
  subcategories_model({
    required this.id,
    required this.title,
    required this.image,
    required this.categories_id
  });
  factory subcategories_model.fromjson(Map<String, dynamic> json) {
    return subcategories_model(
      id: json['id'],
      title: json['title'],
      image: json['image'],
      categories_id: json['categories_id'],
    );
  }
}

File: movie_back/app/Models/subcategories_model.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class subcategories_model extends Model
{
    use HasFactory;
    protected $table="subcategories";
    protected $primaryKey = "id";
    protected $fillable = [
        "title ",
        "image,",
        'categories_id ',
        "status",
    ];
    public function categories(){
        return $this->belongsTo(categories_model::class);
    }
    public function movie(){
        return $this->hasMany(movie_model::class,'movie_id');
    }
}
